### PR TITLE
[tensor] refactor chunk mgr and impl MemStatsCollectorV2

### PR DESCRIPTION
**Pull request by ver217**
Refactor chunk mgr for easily monitoring memory usage.

### CODE REVIEW

File: colossalai/tensor/chunk.py [hunk 1]
```diff
@@ -236,10 +235,9 @@ def access_chunk(self, tensor: torch.Tensor) -> None:
         self.accessed_chunks.add(chunk)
         self.total_mem[chunk.device_type] += chunk.mem
 
-    def release_chunk(self, tensor: torch.Tensor) -> None:
+    def release_chunk(self, chunk: Chunk) -> None:
```
Comment (feifeibear) [inline colossalai/tensor/chunk.py:238]: will the API changing affects our old code?

File: colossalai/tensor/chunk.py [hunk 2]
```diff
@@ -340,3 +332,17 @@ def copy_chunk_group(self, dest_group_name: str, src_group_name: str):
         for dest_chunk, src_chunk in zip(self.chunk_groups[dest_group_name], self.chunk_groups[src_group_name]):
             if not dest_chunk.is_free:
                 dest_chunk.copy_(src_chunk)
+
+    def get_chunks(self, tensors: Iterable[torch.Tensor]) -> FrozenSet[Chunk]:
+        return frozenset([self.get_chunk(tensor) for tensor in tensors])
+
+    def add_extern_static_tensor(self, tensor: torch.Tensor) -> None:
```
Comment (feifeibear) [inline colossalai/tensor/chunk.py:339]: I think you need to make sure the static tensor is not registered as a chunk managed tensor later.

Comment (ver217) [reply colossalai/tensor/chunk.py:339]: Yes. Done.

File: colossalai/tensor/chunk.py (hunk 1 above)
Comment (ver217) [reply colossalai/tensor/chunk.py:238]: Chunk manager's methods should not be called by users directly. I think user's code won't be influenced by this PR.

**Review by feifeibear — APPROVED** [review]

**Pull request merged**